src/sentry/notifications/notifications/strategies/role_based_strategy.py

```diff
@@ -0,0 +1,62 @@
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, Iterable, MutableMapping
+
+from sentry import roles
+from sentry.models import OrganizationMember, Team
+
+if TYPE_CHECKING:
+    from sentry.models import Organization, User
+
+
+class RoleBasedStrategy:
```

Comment:
is this an ABC?

Reply:
Yep, guess I should inherit from it

Reply:
you can use the `@abstractmethod` decorator below if you do that.